server/appender.go

```diff
@@ -0,0 +1,572 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"math"
+	"sync"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/pkg/errors"
+	"github.com/prometheus/prometheus/model/exemplar"
+	"github.com/prometheus/prometheus/model/histogram"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/model/metadata"
+	"github.com/prometheus/prometheus/model/textparse"
+	"github.com/prometheus/prometheus/model/timestamp"
+	"github.com/prometheus/prometheus/model/value"
+	"github.com/prometheus/prometheus/scrape"
+	"github.com/prometheus/prometheus/storage"
+)
+
+var errNameLabelMandatory = fmt.Errorf("missing metric name (%s label)", labels.MetricName)
+
+type appendErrors struct {
+	numOutOfOrder         int
+	numDuplicates         int
+	numOutOfBounds        int
+	numExemplarOutOfOrder int
+}
+
+// appendFn an scrapecache are copied from prometheus project. Ideally it should've been used from the public API.
+// The public API allows using the ScrapeManager only. We can later use scrape.ScrapeManager but for
+// now we are only using the bits necessary for the project.
+func appendFn(app storage.Appender, logger log.Logger, cache *scrapeCache, b []byte, contentType string, ts time.Time, conf *configuration) error {
+	p, err := textparse.New(b, contentType, true)
+	if err != nil {
+		_ = level.Debug(logger).Log(
```

Comment:
Might be worth adding a comment here that `textparse.New` _always_ returns a valid parser. Therefore we continue even if there's an error. Pretty bad API design if you tell me, but we move on :)